lib/decompress/huf_decompress.c

```diff
@@ -270,6 +278,16 @@ HUF_decompress1X1_usingDTable_internal_body(
     return dstSize;
 }
 
+MEM_STATIC int BIT_reloadDStreamFast(BIT_DStream_t* bitD)
```

Comment:
This helps gcc, but is surprisingly neutral for clang.

The idea is that we don't need to include all the end of stream code in the main loop body, since we have the completion calls at the end anyways. This removes branches and code from the large hot loop.